File: packages/javascript/src/core/severity.ts
/**
 * Test severity levels - compatible with Allure.
 */

export enum Severity {
  BLOCKER = 'blocker',
  CRITICAL = 'critical',
  NORMAL = 'normal',
  MINOR = 'minor',
  TRIVIAL = 'trivial',
}

/**
 * Convert string to Severity enum.
 */
export function parseSeverity(value: string): Severity {
  const normalized = value.toLowerCase();
  switch (normalized) {
    case 'blocker':
      return Severity.BLOCKER;
    case 'critical':
      return Severity.CRITICAL;
    case 'minor':
      return Severity.MINOR;
    case 'trivial':
      return Severity.TRIVIAL;
    default:
      return Severity.NORMAL;
  }
}

File: packages/python/src/qagentic/core/severity.py
"""Severity levels for test cases."""

from enum import Enum


class Severity(str, Enum):
    """Test severity levels - compatible with Allure."""
    
    BLOCKER = "blocker"
    CRITICAL = "critical"
    NORMAL = "normal"
    MINOR = "minor"
    TRIVIAL = "trivial"
    
    def __str__(self) -> str:
        return self.value
    
    @classmethod
    def from_string(cls, value: str) -> "Severity":
        """Convert string to Severity enum."""
        value_lower = value.lower()
        for severity in cls:
            if severity.value == value_lower:
                return severity
        return cls.NORMAL
